server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/utils/AlignedSeriesCompactionExecutor.java

```diff
@@ -99,9 +105,18 @@ private List<IMeasurementSchema> collectSchemaFromAlignedChunkMetadataList(
           if (chunkMetadata == null) {
             continue;
           }
+          if (measurementSet.contains(chunkMetadata.getMeasurementUid())) {
+            continue;
+          }
+          measurementSet.add(chunkMetadata.getMeasurementUid());
+          Chunk chunk = reader.readMemChunk((ChunkMetadata) chunkMetadata);
+          ChunkHeader header = chunk.getHeader();
           schemaSet.add(
               new MeasurementSchema(
-                  chunkMetadata.getMeasurementUid(), chunkMetadata.getDataType()));
+                  header.getMeasurementID(),
+                  header.getDataType(),
+                  header.getEncodingType(),
+                  header.getCompressionType()));
```

Comment:
The chunk's data seems to be wasted. Is there any chance to reuse it? Maybe you can retrieve it from the chunk cache.

Reply:
resolve